fedimint-core/src/bitcoin_migration.rs

```diff
@@ -0,0 +1,194 @@
+use std::str::FromStr;
+
+use bitcoin_hashes::hex::FromHex;
+
+pub fn bitcoin29_to_bitcoin30_public_key(pk: bitcoin::PublicKey) -> bitcoin30::PublicKey {
+    bitcoin30::PublicKey::from_slice(&pk.to_bytes()).unwrap()
+}
+
+pub fn bitcoin29_to_bitcoin30_network(network: bitcoin::Network) -> bitcoin30::Network {
+    match network {
+        bitcoin::Network::Bitcoin => bitcoin30::Network::Bitcoin,
+        bitcoin::Network::Testnet => bitcoin30::Network::Testnet,
+        bitcoin::Network::Signet => bitcoin30::Network::Signet,
+        bitcoin::Network::Regtest => bitcoin30::Network::Regtest,
+    }
+}
+
+pub fn bitcoin30_to_bitcoin29_script(script: bitcoin30::ScriptBuf) -> bitcoin::Script {
+    bitcoin::Script::from_hex(&script.to_hex_string()).unwrap()
+}
+
+pub fn bitcoin30_to_bitcoin29_address(address: bitcoin30::Address) -> bitcoin::Address {
+    bitcoin::Address::from_str(&address.to_string()).unwrap()
+}
+
+pub fn bitcoin29_to_bitcoin30_ripemd160_hash(
+    hash: bitcoin::hashes::ripemd160::Hash,
+) -> bitcoin30::hashes::ripemd160::Hash {
+    bitcoin30::hashes::ripemd160::Hash::from_str(&hash.to_string()).unwrap()
```

Comment:
I got distracted yesterday trying to figure it out myself, so just leaving the question: could we just go `Hash -> &[u8] -> Hash` instead? A typical pattern in `bitcoin_hashes` was to use `&hash[..]` to get the byte representation as `&[u8]` (idk if there's an even better way now). But serializing and deserializing looks very wasteful.

Reply:
Good point! Done.